src/porepy/models/energy_balance.py

```diff
@@ -633,70 +745,115 @@ def bc_type_fourier(self, sd: pp.Grid) -> pp.BoundaryCondition:
         return pp.BoundaryCondition(sd, boundary_faces, "dir")
 
     def bc_type_enthalpy(self, sd: pp.Grid) -> pp.BoundaryCondition:
-        """Dirichlet conditions on all external boundaries.
+        """Boundary conditions on all external boundaries for the advective flux in the
+        energy equation.
+
+        Warning:
+            This must not be different from mixed in definition of
+            :meth:`~porepy.models.fluid_mass_balance.BoundaryConditionsSinglePhaseFlow.
+            bc_type_darcy`.
+            Advective enthalpy flux is due to energy associated with mass entering the
+            system, which is the same as for the mass balance equation.
+            Unphysical systems arise otherwise.
```

Comment:
I think this is important to address. And I also feel like there is some "unnatural" flexibility in how the user can set BC conditions and values, which cause confusion and unphysical systems if not done correctly.